### ISSUE TITLE
Feature Request: Custom Module Names

### ISSUE THREAD

**Issue by sbowler** [Kocal/jsdoc-vuejs#83]
Thanks for your work on this. One problem I have is I'd like to be able to differantiate my ES2015 modules from my vue component "modules" as documented. One idea to help with this I think might be to allow us to name a module in a .vue file to override using the file name. At least that way I can call it UserComponent in the documentation or something similar rather than just User if it's called User.vue. 

I'm open to other ideas or suggestions. Ideally there'd be a good jsdoc way to separate them, but that may be more difficult or may involve more template editing I guess.

**Comment by Kocal** [comment]
This feature has _already existed_ in 1.x, but it was a pain in the ass to have something that works ([1st impl.](https://github.com/Kocal/jsdoc-vuejs/blob/51e31ad68d2dbf56ee65c7ad827d6b93a4860d44/lib/vueTag.js), [2nd impl.](https://github.com/Kocal/jsdoc-vuejs/blob/6c40a9846a563573b27ba55331659d60bed5dce9/lib/vueTag.js), [3rd impl.](https://github.com/Kocal/jsdoc-vuejs/blob/ab02d40b3320b4942a8015463bec2e70fbc4e267/lib/vueTag.js), [4th and last impl.](https://github.com/Kocal/jsdoc-vuejs/blob/eb2f4a344d4c6675fdb2d5218b804194ef1623a9/lib/vueTag.js)).

Since the source code is cleaner and I'm more comfortable with JSDoc, I think we can came up to a clean and functional feature.

**Comment by linkdd** [comment]
I managed to get it working by:

 - moving those lines : [index.js L38-L41](https://github.com/Kocal/jsdoc-vuejs/blob/master/index.js#L38-L41) inside [this if statement](https://github.com/Kocal/jsdoc-vuejs/blob/master/index.js#L26)
 - using the following doclet : 

```javascript
/**
 * Description
 * @module MyCustomModule
 * @vue-data {Boolean} foo Some description
 */
```

The current behavior is used whenever the ``@module`` tag is not used.

However, it doesn't work if this tag isn't in the same doclet : 

```javascript
/**
 * Description
 * @module MyCustomModule
 */

/**
 * @vue-data {Boolean} foo bar
 */
```

Is it possible to include this implementation ?

**Comment by Kocal** [comment]
What is the behavior if you have a file named `MyCustomModule.js` that exports a class `MyCustomModule`?

I started to work on a implementation that will suffix `.vue` to namespace of Vue components (#87), but conflict can happens if you have `MyCustomModule.js` and `MyCustomModule.vue` files :thinking:

**Comment by linkdd** [comment]
Just tried it out, it seems both documentation are merged in a single page (thanks to the ``e.doclet.description || ''`` at line 42).

I have split my code into a hierarchy (``layouts``, ``components`` and ``pages`` folders) and I wanted to see them in the documentation like this:

 - ``layouts/Main``
 - ``components/Form``
 - ...

I copied the ``index.js`` inside my projet with my modifications, and I'm using it while the feature is not upstream

**Comment by Kocal** [comment]
> One idea to help with this I think might be to allow us to name a module in a .vue file to override using the file name.

Let's implement this and write specs! :skull_and_crossbones: 
I've planned to work on it this evening, but feel free to open a PR :slightly_smiling_face:

**Comment by Kocal** [comment]
Released in [v2.1.0](https://github.com/Kocal/jsdoc-vuejs/blob/master/CHANGELOG.md#210-2018-07-06) :tada:
